components/ImageGallery.js

```diff
@@ -0,0 +1,356 @@
+      // We want to allow switching between gallery images only if
+      // the image is at its left of right boundary. This happens if the
+      // image is fully zoomed out, or if the image is zoomed in but the
+      // user moved it to one of its boundaries.
+      this.setState({
+        imageSwitchingEnabled: imageBoundaryReached,
+      });
+    }
+  }
+
+  setImagePreviewMode() {
+    const { onModeChanged } = this.props;
+
+    this.setState({ mode: IMAGE_PREVIEW_MODE });
+
+    this.timingDriver.runTimer(1, () => {
+      if (_.isFunction(onModeChanged)) {
+        onModeChanged(IMAGE_PREVIEW_MODE);
+      }
+    });
+  }
+
+  setGalleryMode() {
+    const { onModeChanged } = this.props;
+
+    this.setState({ mode: IMAGE_GALLERY_MODE });
+
+    this.timingDriver.runTimer(0, () => {
+      if (_.isFunction(onModeChanged)) {
+        onModeChanged(IMAGE_GALLERY_MODE);
+      }
+    });
+  }
+
+  onViewTransformed(event) {
+    const { mode } = this.state;
+
+    if (event.scale > 1.0 && mode === IMAGE_GALLERY_MODE) {
+      // If controls are visible and image is transformed,
+      // We should switch to image preview mode
+      this.setImagePreviewMode();
+    } else if (mode === IMAGE_PREVIEW_MODE) {
+      this.updateImageSwitchingStatus();
+    }
+  }
+
+  onSingleTapConfirmed() {
+    const { mode } = this.state;
+
+    if (mode === IMAGE_PREVIEW_MODE) {
+      // If controls are not visible and user taps on image
+      // We should switch to gallery mode
+      this.setGalleryMode();
+    } else {
+      this.setImagePreviewMode();
+    }
+  }
+
+  renderDescription(description) {
+    const { collapsed } = this.state;
+    const { style } = this.props;
+
+    if (!description) return;
+
+    const descriptionIcon = collapsed ? <Icon name="up-arrow" /> : <Icon name="down-arrow" />;
+
+    const descriptionText = (
+      <Subtitle
+        style={style.description.text}
+        numberOfLines={collapsed ? 2 : null}
+      >
+        {description}
+      </Subtitle>
+    );
+
+    return (
+      <View
+        style={style.description.container}
+        driver={this.timingDriver}
+        animationName="fadeOut"
+      >
+        <TouchableOpacity onPress={collapsed ? this.collapseDescription : this.expandDescription}>
+          {description.length >= DESCRIPTION_LENGTH_TRIM_LIMIT ? descriptionIcon : null}
+        </TouchableOpacity>
+        <ScrollView
+          onScroll={this.onDescriptionScroll}
+          scrollEventThrottle={200}
+          style={style.description.scroll}
+        >
+          {descriptionText}
+        </ScrollView>
+      </View>
+    );
+  }
+
+  renderTitle(title) {
+    const { style } = this.props;
+
+    return (
+      <View
+        style={style.title.container}
+        driver={this.timingDriver}
+        animationName="fadeOut"
+      >
+        <Subtitle style={style.title.text} numberOfLines={2}>{title}</Subtitle>
+      </View>
+    );
+  }
+
+  getImageTransformer(page) {
```

Comment:
Can you keep render methods grouped together, it is much cleaner.

You can also group, update methods, get methods and other.